src/Storages/MergeTree/MergeTreeIndexText.cpp

```diff
@@ -759,6 +759,21 @@ void MergeTreeIndexGranuleTextWritable::deserializeBinary(ReadBuffer &, MergeTre
     throw Exception(ErrorCodes::LOGICAL_ERROR, "Deserialization of MergeTreeIndexGranuleTextWritable is not implemented");
 }
 
+size_t MergeTreeIndexGranuleTextWritable::memoryUsageBytes() const
+{
+    size_t posting_lists_size = 0;
+    for (auto plist : posting_lists)
+        posting_lists_size += plist.getSizeInBytes();
+
+    return sizeof(*this)
+        + bloom_filter.getFilterSizeBytes()
+        /// can ignore the sizeof(PostingListBuilder) here since it is just references to tokens_map
+        + tokens_and_postings.capacity() * sizeof(std::pair<StringRef, PostingListBuilder *>)
+        + tokens_map.getBufferSizeInBytes()
+        + posting_lists_size
+        + arena->allocatedBytes();
```

Comment:
seems `arena` is used for allocate space for keys of `tokens_map` https://github.com/ClickHouse/ClickHouse/blob/ef11ae76f7b0be10fcf4148f7fbbfcafcd5a8002/src/Storages/MergeTree/MergeTreeIndexText.cpp#L830-L831
Should we just use arena here instead of tokens_map?

Reply:
Does anyone know where this memoryUsageBytes() is actually called? I can't find an actual call anywhere...

Reply:
Some of these tables are virtual tables not really stored anywhere. What they show is generated dynamically lazyly when you call a `select` on them.

IIRC the `system.parts` has the `secondary_indices` and other size fields that (in principle) could need this function. If none of them is using this function at any point; then the index size we are reporting is probably wrong and we need to check why.

Reply:
> Worst case is it is a slight overestimate, which I think is OK

yes, that's fine it does not have to accurate. Even thought it's not used I don't see any harm to have it.